### PR TITLE
feat: fetch all NFTs ( with tokenStandard)  by collection address

### PR DESCRIPTION
**Pull request by urbentom**
Created new function `fetchAllDigitalAssetByVerifiedCollection`  which will run several gpa calls to try and fetch all the NFTs by their verified collection. This method only works on newer collections that have their tokenStandard set.

### CODE REVIEW

File: clients/js/test/digitalAsset.test.ts [hunk 1]
```diff
@@ -231,6 +235,69 @@ test('it can fetch all Metadata accounts by owner', async (t) => {
...
+
+  const collectionB = await createDigitalAssetWithToken(umi, {
+    isCollection: true,
+    authority: collectionAuthority,
+  });
+
+  const mintA1 = await createDigitalAssetWithVerifiedCollection(umi, {
+    collection: {
+      key: collectionA.publicKey,
+      verified: false,
```
Comment (febo) [inline clients/js/test/digitalAsset.test.ts:255]: Can we 'omit' the `verified` field here?

File: clients/js/test/digitalAsset.test.ts [hunk 2]
```diff
@@ -254,3 +321,24 @@ function createDigitalAssetWithSecondCreator(
     ]),
   });
 }
+
+async function createDigitalAssetWithVerifiedCollection(
+  context: Parameters<typeof createDigitalAssetWithToken>[0],
+  input: Parameters<typeof createDigitalAssetWithToken>[1] & {
```
Comment (febo) [inline clients/js/test/digitalAsset.test.ts:327]: Can we 'omit' the verified field here so we don't need to pass `false` all the time?

File: clients/js/src/digitalAsset.ts [hunk 3]
```diff
@@ -115,6 +118,32 @@ export async function fetchAllDigitalAssetByCreator(
...
+    COLLECTION_OFFSETS.map(async (offset) => {
+      try {
+        return await getMetadataGpaBuilder(context)
+          .where(offset, 1)
+          .where(offset + VERIFIED_COLLECTION_OFFSET, 1)
+          .where(offset + COLLECTION_ADDRESS_OFFSET, collectionAddress)
+          .sliceField('mint')
+          .getDataAsPublicKeys();
+      } catch (e) {
+        return [];
```
Comment (febo) [inline clients/js/src/digitalAsset.ts:139]: Do we need to bubble the error to the client? E.g., if there is a connection error.

**Review by febo — CHANGES_REQUESTED** [review]
Looks good, left some minor comments. Also, I think it would be good to add a test with a different number of creators on each NFT, so we know that the offset is working.

File: clients/js/src/digitalAsset.ts (hunk 3 above)
Comment (urbentom) [reply clients/js/src/digitalAsset.ts:139]: Yea good point, i've removed the try / catch

**Review by febo — APPROVED** [review]
LGTM

**Pull request merged**